stark-brainfuck/Cargo.toml

```
@@ -0,0 +1,40 @@
+[package]
+name = "stark-brainfuck"
+version = "0.2.0"
+authors = ["Triton Software AG"]
+edition = "2021"
+
+license = "GPL-2.0"
+description = "A Turing-complete zk-STARK engine consisting of a virtual machine, prover and verifier."
+homepage = "https://github.com/Neptune-Crypto/twenty-first"
+documentation = "https://github.com/Neptune-Crypto/twenty-first"
+repository = "https://github.com/Neptune-Crypto/twenty-first"
+readme = "../README.md"
+
+keywords = ["post-quantum", "algebra", "rescue-prime"]
+categories = ["cryptography", "mathematics"]
+
+[dev-dependencies]
+proptest = "1.0"
+
+[dev-dependencies.criterion]
+version = "0.3"
+features = ["html_reports"]
+
+[dev-dependencies.cargo-husky]
+version = "1"
+default-features = false
+features = ["precommit-hook", "run-cargo-clippy", "run-cargo-fmt"]
+
+[dependencies]
+twenty-first = { path = "../twenty-first" }
```

Why not the crates.io crate?

---

Because stark-brainfuck functions as a test suite for twenty-first. If we depend on the crates.io version, running `make` or `cargo ...` will not trigger failure when a branch makes a breaking change. We would like to know that twenty-first diverged from being compatible with stark-brainfuck before it hits crates.io.

Related: Making twenty-first a path dependency only will prevent stark-brainfuck from being uploaded to crates.io. If we should want to do that, it is possible to [specify both `path` and `version`](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations). This requires bumping the version number in another location, though:

```suggestion
twenty-first = { path = "../twenty-first", version = "0.2.0" }
```